analytics.md

```diff
@@ -0,0 +1,54 @@
+ - Hit Type `t` - The type of analytics hit, `faas-cli` uses the `event` type.
+   - https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#t
+ - Client ID `cid` - A `faas-cli` analytics user ID, e.g. `c1481af7-8682-462f-a586-71d752dbe87b`. This is a UUID4 generated by the [`github.com/satori/go.uuid`](https://github.com/satori/go.uuid) package and stored in the local config file `~/.openfaas/analytics_uuid`. This does not allow us to track individual users but does enable us to accurately measure user counts vs. event counts.
+   - https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#cid 
+ - ApplicationName `an` - The application name, e.g. `faas-cli`.
+   - https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#an
+ - ApplicationVersion `av` - The `faas-cli` version, e.g. `0.4.18d`.
+   - https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#av
+ - Anonymize IP `aip` - The IP address will be anonymized when submitting the analytics event, e.g. `1`.
+   - https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#aip
+ - Language `cd1` - A Custom Dimension containing the function language for `new`, `build` and `deploy` commands (returns `unset` for all other subcommands). 
+   - https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#cd_
+ - Operating System `cd2` - A Custom Dimension containing the OS on which `faas-cli` is being run, this is determined from `runtime.GOOS`.
+   - https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#cd_
+ - Architecture `cd3` - A Custom Dimension containing the Architecture on which `faas-cli` is being run, this is determined from `runtime.GOOS`.
+   - https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#cd_
+- Event Category `ec` - The `faas-cli` groups all successful CLI actions under the `cli-success` category, and failures (not yet implemented) under `cli-failure`.
+   - https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#ec
+- Event Action `ea` - Each command invoked by the user is results in an analytics event being generated with the same name as the `faas-cli` subcommand.
+   - https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#ea
+
+## How?
+
+The core implementation is in the `analytics` package, with calls to `analytics.Event()` each time a subcommand is invoked.
+
+Analytics submission occurs in a background goroutine and will fail fast to avoid delaying any execution.
+
+Due to the speed with which most OpenFaaS API calls return, a timeout of 300 milliseconds is used to give the analytics submission time to complete.
+
+## Opting out?
+
+If you wish to opt out of submitting analytics events you may do so by setting the `OPEN_FAAS_TELEMETRY` environment variable.
+
+    export OPEN_FAAS_TELEMETRY=0
```

Comment:
Our NS should be `OPENFAAS_`

Reply:
Done

Reply:
I can't see where it was changed? The code review doesn't say "see outdated".. did you push yet?

Reply:
Missed pushing the README it seems.